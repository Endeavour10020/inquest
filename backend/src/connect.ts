// needed for typeorm && type-graphl to function
import "reflect-metadata";
import { ApolloServer } from "apollo-server";
import { Container } from "typedi";
import * as TypeORM from "typeorm";
import * as TypeGraphQL from "type-graphql";
import { ALL_ENTITIES } from "./entities";
import { ALL_RESOLVERS } from "./resolvers";
import { Context } from "./context";
import { seedDatabase } from "./helpers";

// register 3rd party IOC container

export async function createSQLiteServer() {
    // create TypeORM connection
    TypeORM.useContainer(Container);
    await TypeORM.createConnection({
        type: "sqlite",
        database: ":memory:",
        entities: ALL_ENTITIES,
        synchronize: true,
        logger: "debug",
        cache: true,
    });

    // seed database with some data
    const { defaultUser } = await seedDatabase();

    // build TypeGraphQL executable schema
    const schema = await TypeGraphQL.buildSchema({
        resolvers: ALL_RESOLVERS,
        container: Container,
    });

    // create mocked context
    const context: Context = { user: defaultUser };

    // Create GraphQL server
    return new ApolloServer({ schema, context });
}
