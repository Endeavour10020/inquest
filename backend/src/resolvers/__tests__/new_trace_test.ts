import { createSQLiteServer } from "./../../connect";
import { ApolloServer } from "apollo-server";
import { Container } from "typedi";
import {
    createTestClient,
    ApolloServerTestClient,
} from "apollo-server-testing";
import gql from "graphql-tag";

const NEW_TRACE = gql`
    mutation newTrace(
        $module: String!
        $function: String!
        $statement: String!
    ) {
        newTrace(
            newTraceInput: {
                module: $module
                function: $function
                statement: $statement
            }
        ) {
            module
        }
    }
`;

let server: ApolloServer;
let client: ApolloServerTestClient;
beforeAll(async () => {
    Container.reset();
    server = await createSQLiteServer();
    client = createTestClient(server);
});

afterAll(() => {
    server.stop();
    Container.reset();
});

it("should create new trace object", async () => {
    expect(
        await client.mutate({
            mutation: NEW_TRACE,
            variables: {
                module: "mod",
                function: "func",
                statement: "statement",
            },
        })
    ).toMatchSnapshot();
});
