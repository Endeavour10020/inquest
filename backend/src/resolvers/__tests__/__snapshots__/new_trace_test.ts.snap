// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries test new trace subscription 1`] = `
Object {
  "data": Object {
    "newTrace": Object {
      "module": "mod",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
